include/seqcmd.h

```diff
@@ -0,0 +1,510 @@
+
+/**
+ * Set the volume of a specific channel on a given seqPlayer over a specified duration
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param channelIndex the index of the channel to modify
+ * @param duration the duration to set the volume over
+ * @param volume the target volume for the sequence
+ */
+#define SEQCMD_SET_CHANNEL_VOLUME(seqPlayerIndex, channelIndex, duration, volume)                                    \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_CHANNEL_VOLUME << 28) | ((u8)(seqPlayerIndex) << 24) | ((u8)(duration) << 16) | \
+                      ((u8)(channelIndex) << 8) | ((u8)volume))
+
+/**
+ * Write a value to an array that can be read as input directly by the sequence itself. This will be set to the global
+ * ioPort, which can affect the entire sequence. How the sequence responds this io value will entirely depend
+ * on the customized sequence script for each sequence.
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to write the input to
+ * @param ioPort the index of the array to store the input-output value,
+ * @param ioData the value that's written to the input-output array
+ *
+ * @note Each seqPlayer has 8 global ioPorts indexed 0-7.
+ * ioPort 0 and 1 are read-only-once, and will reset after being read.
+ * ioPort 2-7 can be read multiple times.
+ */
+#define SEQCMD_SET_PLAYER_IO(seqPlayerIndex, ioPort, ioData)                                                  \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_PLAYER_IO << 28) | ((u8)(seqPlayerIndex) << 24) | ((u8)(ioPort) << 16) | \
+                      (u8)(ioData))
+
+/**
+ * Write a value to an array that can be read as input directly by the sequence itself. This will be set to the channel
+ * ioPort, which will only affect a single channel. How the sequence responds this io value will entirely depend
+ * on the customized sequence script for each sequence.
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to write the input to
+ * @param channelIndex the index of the channel to write the input to
+ * @param ioPort the index of the array to store the input-output value
+ * @param ioData the value that's written to the input-output array
+ *
+ * @note Each channel has 8 channel ioPorts indexed 0-7.
+ * ioPort 0 and 1 are read-only-once, and will reset after being read.
+ * ioPort 2-7 can be read multiple times.
+ */
+#define SEQCMD_SET_CHANNEL_IO(seqPlayerIndex, channelIndex, ioPort, ioData)                                    \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_CHANNEL_IO << 28) | ((u8)(seqPlayerIndex) << 24) | ((u8)(ioPort) << 16) | \
+                      ((u8)(channelIndex) << 8) | (u8)(ioData))
+
+/**
+ * Disable (or reenable) specific channels from receiving input set by `SEQCMD_SET_CHANNEL_IO`
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param channelMask a u16 bitmask where each of the 16 bits mapped to one of the 16 channel. Bitflag on to disable
+ *
```

Comment:
feel like this can be simplified to "a 16 bit mask where each bit maps to a channel". applies to a few places